### DISCUSSION TITLE
Support for database persistence layer

### DISCUSSION THREAD

**Discussion by atymic** [spatie/laravel-webhook-server#91] [Ideas]
I know this is kind of a departure of how this package currently works (fully queue based) but unfortunately we can't use queue delays past 15 minutes.

As a result, we'd like to switch to using a database persistence layer. It also has some other advantages, like being able to debug failed webhooks much easier.

Understand if you don't want it in this package, just though i'd check :)

Flow: 
- Create webhook as normal
- Insert record into DB
- Dispatched to queue for initial dispatch
- If fail, set the attempt count and the next retry time on the model
- Cron picks up ready tries after the first try and dispatches them to the queue

**Comment by freekmurze** [comment]
I'm going to leave things as they are for now, but thank you for your suggestion.

**Comment by siarheipashkevich** [comment]
@atymic did you implement you own solution?

**Reply by atymic** [reply]
Yep i did, running smoothly with millions of webhooks :)

**Reply by siarheipashkevich** [reply]
Did you use `events` which fired the package or you forked and modified package?

**Reply by atymic** [reply]
I wrote my own implementation from scratch inspired by the package.
